docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+### Required resolvers
+
+To support the enhanced introspection query, a subgraph service must define resolvers for the following fields:
+
+- `Query._service`
+- `_Service.sdl`
+
+`Query._service` returns a `_Service` object, which in turn has a single field, `sdl` (short for schema definition language). The `sdl` field returns a string representation of the subgraph's schema.
+
+The returned `sdl` string has the following requirements:
+
+* It must **include** all uses of all federation-specific directives, such as `@key`.
+    * For a list of all federation-specific subgraph directives, see the [glossary](#glossary-of-schema-additions).
+* It must **omit** all automatically added definitions from the [subgraph schema specification](#subgraph-schema-specification), such as `Query._service` and `_Service.sdl`!
+
+For example, consider this Federation 2 subgraph schema:
+
+```graphql title="schema.graphql"
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: ["@key"])
+
+type Query {
+  me: User
+}
+
+type User @key(fields: "id") {
+  id: ID!
+}
+```
+
+The value returned for the `sdl` field should include all of this information, including directives (excessive whitespace can be removed).
+
+## Generating stub types
+
+Each subgraph in a supergraph should be able to run successfully in isolation, regardless of whether any _other_ subgraphs are running. Many subgraphs run successfully in isolation with no additional logic required. However, subgraph fields _can_ depend on types and fields that are defined in _other_ subgraphs. Whenever this is the case, a subgraph needs to define its own **stubs** for these external types and fields.
+
+To indicate a dependency on another subgraph's type or field, a subgraph schema field uses either the `@requires` or `@provides` directive:
+
+```graphql title="Shipping subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  weight: Int! @external
+  shippingCost: Int! @requires("weight")
+}
+```
+
+In the above example, the Shipping subgraph defines a `shippingCost` field for the `Product` entity. The `shippingCost` field `@requires` the value of the `weight` field, which is a field that's resolved by a _different_ subgraph. To indicate that `weight` is resolved only by _other_ subgraphs, the Shipping subgraph _also_ defines the field, but it applies the `@external` directive to it.
+
+This allows us to be explicit about dependencies on another service, and service composition will verify that an `@external` field matches up with the original field definition, which can catch mistakes or migration issues (when the original field changes its type for example). `@external` fields also give an individual service the type information it needs to validate and decode incoming representations (this is especially important for custom scalars), without requiring the composed graph schema to be available at runtime in each service.
```

Comment:
nit: in fed v1 `@external` was required on any field referenced from `@key/@requires/@provides`, fed v2 relaxed this requirement and now it is only needed for fields referenced by `@requires` directive and in some cases for `@provides` (only needed if field is resolved locally in some but not all paths)